clang-tools-extra/test/clang-tidy/checkers/daedalean-switch-statement.cpp

```diff
@@ -0,0 +1,21 @@
+// RUN: %check_clang_tidy %s daedalean-switch-statement %t
+
+void bar();
+
+void foo(int a) {
+
+  switch (a) {
+  case 1:
+    // CHECK-MESSAGES: :[[@LINE-1]]:9: warning: Case statement must be scoped statement [daedalean-switch-statement]
+  case 2:
+    // CHECK-MESSAGES: :[[@LINE-1]]:9: warning: Case statement must be scoped statement [daedalean-switch-statement]
+    bar();
+    break;
+  case 3: {
+    bar();
+  }
```

Comment:
what about explicit fallthrough? we don't want something like this to be flagged:

```
switch(x) {
case 1:
  [[ fallthrough ]]
case 2: {
    bla();
  }
}
```

Maybe not even without the fallthrough, and then have another check for missing break or fallthrough at the end of a case

Reply:
-Wimplicit-fallthrough compiler flag check exactly this

Reply:
As discussed: fallthrough is a statement, so we handle it as such and require it to be scoped as well.